homeassistant/components/connector/cover.py

```diff
@@ -0,0 +1,295 @@
+    12: CoverDeviceClass.CURTAIN,
+    13: CoverDeviceClass.CURTAIN,
+    14: CoverDeviceClass.CURTAIN,
+    26: CoverDeviceClass.SHADE,
+    43: CoverDeviceClass.SHUTTER,
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Motion Blind from a config entry."""
+    entities = []
+    connector = hass.data[DOMAIN][config_entry.entry_id][KEY_GATEWAY]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+    device_list = await connector.device_list()
+    if device_list is None:
+        return
+    for hub in device_list.values():
+        if hub.devicetype in WIFIMOTORTYPE:
+            wifi_motor_type = POSITION_DEVICE_MAP.get(hub.type, CoverDeviceClass.SHADE)
+            entities.append(
+                TwoWayDevice(
+                    coordinator=coordinator,
+                    blind=hub,
+                    device_class=wifi_motor_type,
+                    config_entry=config_entry,
+                )
+            )
+        else:
+            for blind in hub.blinds_list.values():
+                blind_type = POSITION_DEVICE_MAP.get(blind.type, CoverDeviceClass.SHADE)
+                if blind.wireless_mode in ONEWAYWIRELESSMODE:
+                    entities.append(
+                        OneWayDevice(
+                            coordinator=coordinator,
+                            blind=blind,
+                            device_class=blind_type,
+                            config_entry=config_entry,
+                        )
+                    )
+                elif blind.wireless_mode in TWOWAYWIRELESSMODE:
+                    if blind.type in VENETIANTYPE:
+                        entities.append(
+                            TiltDevice(
+                                coordinator=coordinator,
+                                blind=blind,
+                                device_class=blind_type,
+                                config_entry=config_entry,
+                            )
+                        )
+                    else:
+                        entities.append(
+                            TwoWayDevice(
+                                coordinator=coordinator,
+                                blind=blind,
+                                device_class=blind_type,
+                                config_entry=config_entry,
+                            )
+                        )
+                else:
+                    _LOGGER.info("This wirelessMode not support")
+
+    async_add_entities(entities)
+
+
+class TwoWayDevice(CoordinatorEntity, CoverEntity):
+    """Representation of a Motion Blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry):
+        """Initialize the blind."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+        self._device_class = device_class
+        self._config_entry = config_entry
```

Comment:
You don't need this, it's already in `self.platform.config_entry`

Reply:
It is more convenient for me to obtain it this way, because some data points are wifi curtains, and some are sub-devices under the hub